### PR TITLE
Add toggle for debugProfileWidgetBuilds

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

In https://github.com/flutter/flutter/pull/21492, we added a service extension to toggle profiling of widget build times. This was never wired up through the tool, so it was essentially dead code.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @fkorotkov we're getting some odd git errors in the cirrus jobs for this PR

**Review by dnfield — APPROVED** [review]
LGTM.

Do the IDEs need to know about this, or are they already wired in for it?

**Comment by jonahwilliams** [comment]
cc @devoncarew

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -976,6 +993,7 @@ abstract class ResidentRunner {
         printStatus('To toggle the widget inspector (WidgetsApp.showWidgetInspectorOverride), press "i".');
         printStatus('To toggle the display of construction lines (debugPaintSizeEnabled), press "p".');
         printStatus('To simulate different operating systems, (defaultTargetPlatform), press "o".');
+        printStatus('To enable profiling of widget build times, (debugProfileWidgetBuilds), press "a".');
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/resident_runner.dart]: The dartdoc for `debugProfileWidgetBuilds` is 'Adds [Timeline] events for every Widget built'.

Perhaps, in order to keep this description terse, we say: 'Add timeline events for every widget build method'? Or, 'To enable timeline events for all widget build method, (debugProfileWidgetBuilds), press "a"'?

**Comment by devoncarew** [comment]
We did not expose this though IDEs, but it would be good to do, in order to see how useful it is.

cc @DanTup for VS Code; I'll open an issue for DevTools

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 1 above)
Comment (jacob314) [reply packages/flutter_tools/lib/src/resident_runner.dart]: This makes sense to expose now that we have the timeline page in devtools to display the data right next to the button to toggle it on and off.

**Comment by DanTup** [comment]
@devoncarew @dnfield I've wired it up, but I don't know what it's supposed to do so I'm not sure how to verify it's working. I don't see any  difference in the DevTools timeline view, nor the Logging view (though I can see it being toggled on/off). Where does it show up?

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Done

**Comment by Hixie** [comment]
This change is fine, but FWIW, we should be wary of adding things like this just for completeness. We don't necessarily want every last feature exposed in the UI. It will make it unusable.

**Comment by jonahwilliams** [comment]
We ended up using this to debug an internal issue, so its usefulness justified the cost IMO.

**Pull request merged**